protocol/contracts/beanstalk/sun/SeasonFacet/Weather.sol

```diff
@@ -168,18 +178,98 @@ contract Weather is Sun {
             return;
         } else if (!s.season.raining) {
             s.season.raining = true;
+            address[] memory wells = LibWhitelistedTokens.getWhitelistedWellLpTokens();
             // Set the plenty per root equal to previous rain start.
-            s.sops[s.season.current] = s.sops[s.season.rainStart];
+            uint32 season = s.season.current;
+            uint32 rainstartSeason = s.season.rainStart;
+            for (uint i; i < wells.length; i++) {
+                s.sops[season][wells[i]] = s.sops[rainstartSeason][wells[i]];
+            }
             s.season.rainStart = s.season.current;
             s.r.pods = s.f.pods;
             s.r.roots = s.s.roots;
         } else {
+            // flood podline first, because it checks current Bean supply
+            floodPodline();
+
             if (s.r.roots > 0) {
-                // initalize sopWell if it is not already set.
-                if (s.sopWell == address(0)) s.sopWell = well;
-                sop();
+                WellDeltaB[] memory wellDeltaBs = getWellsByDeltaB();
+                wellDeltaBs = calculateSopPerWell(wellDeltaBs);
+
+                for (uint i; i < wellDeltaBs.length; i++) {
+                    sop(wellDeltaBs[i]);
+                }
+            }
+        }
+    }
+
+    function floodPodline() private {
+        // Make 0.1% of the total bean supply worth of pods harvestable.
+
+        uint256 totalBeanSupply = C.bean().totalSupply();
+        uint256 sopFieldBeans = totalBeanSupply.div(FLOOD_PODLINE_PERCENT_DENOMINATOR); // 1/1000 = 0.1% of total supply
+
+        // Note there may be cases where zero harvestable pods are available. For clarity, the code will still emit an event
+        // but with zero sop field beans.
+        uint256 maxHarvestable = s.f.pods.sub(s.f.harvestable);
+        sopFieldBeans = sopFieldBeans > maxHarvestable ? maxHarvestable : sopFieldBeans;
+
+        s.f.harvestable = s.f.harvestable.add(sopFieldBeans);
+        C.bean().mint(address(this), sopFieldBeans);
+
+        emit SeasonOfPlentyField(sopFieldBeans);
+    }
+
+    function getWellsByDeltaB() public view returns (WellDeltaB[] memory) {
+        address[] memory wells = LibWhitelistedTokens.getWhitelistedWellLpTokens();
+        WellDeltaB[] memory wellDeltaBs = new WellDeltaB[](wells.length);
+        for (uint i = 0; i < wells.length; i++) {
+            wellDeltaBs[i] = WellDeltaB(wells[i], LibWellMinting.cappedReservesDeltaB(wells[i]), 0);
```

Comment:
Make sense. Dragged in some code from gen convert here (to get current deltaB), upon final merge it might have to be de-duplicated if it doesn't merge automagically. https://github.com/BeanstalkFarms/Beanstalk/commit/156a9fc013098aaf479549db3247b54ec90ec0aa

Reply:
lgtm!